### PR TITLE
#30 - Submitting suggestions does not work as expected - Appolin Semegni Fotso

### LINKED ISSUE #30: BUG/2: Submitting suggestions don't work as expected
When submitting a suggestion one expects the following two to happen:
1. We send the suggestion to the backend storing it in the database
2. We move on to the next suggestion.

Currently neither of the above two action is working, so this should be addressed

### PR DESCRIPTION
**Pull request by AppolinFotso**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

I Added a submitButton function and useState in SubmitSuggestion component and I passed both the function and state as props to the PopUpAlert component. The submitButton function, posts data to the server and the state prop stores the return message from the server.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #30

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: client/src/components/PopUpAlert.js [hunk 1]
```diff
@@ -25,12 +25,15 @@ export default function PopUpAlert({ text }) {
...
+      <Button color="success" variant="contained" onClick={() => {
+        props.submitButton();
+        handleClick();
+      }}>
+        {props.text}
       </Button>
       <Snackbar open={open} autoHideDuration={6000} onClose={handleClose}>
         <Alert onClose={handleClose} severity="success" sx={{ width: "100%" }}>
-          Suggestion saved successfully.
+          {props.message === "Suggestions saved successfully" ? props.message : "Unprocessable Entry"}
```
Comment (sztupy) [inline client/src/components/PopUpAlert.js]: Not sure if I understand what this line is trying to do

Comment (AppolinFotso) [reply client/src/components/PopUpAlert.js]: > Not sure if I understand what this line is trying to do

This line checks the message returned by the POST request and displays the correct pop-up.

Comment (sztupy) [reply client/src/components/PopUpAlert.js]: If the idea is to make the error message more user friendly something like "Could not save suggestion" is defintely more useful than "Unprocessable entity" which would not really mean anything to an average user. Also the class above `severity` is set to `sucess` both ways, so even on errors you would receive a green box

Comment (AppolinFotso) [reply client/src/components/PopUpAlert.js]: I added another Alert with **"severity = error"** to handle unsaved suggestion.

File: client/src/components/PopUpAlert.js [hunk 2]
```diff
@@ -24,15 +24,36 @@ export default function PopUpAlert({ text }) {
...
+					>
+						Suggestions saved successfully
+					</Alert>
+				) : (
+					<Alert
+						onClose={handleClose}
+						severity="error"
+						sx={{ width: "100%" }}
+					>
+						Could not saved suggestion
```
Comment (ali-nasir-ali) [inline client/src/components/PopUpAlert.js:53]: This is an excellent use of message here to inform the user about network or database errors.

**Review by ali-nasir-ali — COMMENTED** [review]
clean code and a great thing to clean up the SubmitSuggesstion.js component.

**Comment by sztupy** [comment]
Tested locally, seems to fix the issue at hand

**Pull request merged**